sky/skypilot_config.py

```diff
@@ -306,7 +344,26 @@ def _reload_config_from_internal_file(internal_config_path: str) -> None:
     _loaded_config_path = config_path
 
 
-def _reload_config_hierarchical() -> None:
+def _reload_config_as_server() -> None:
+    global _dict
+    # Reset the global variables, to avoid using stale values.
+    _dict = config_utils.Config()
+
+    overrides: List[config_utils.Config] = []
+    server_config = get_server_config()
+    if server_config:
+        overrides.append(server_config)
+
+    # layer the configs on top of each other based on priority
+    overlaid_server_config: config_utils.Config = config_utils.Config()
+    for override in overrides:
+        overlaid_server_config = overlay_skypilot_config(
+            original_config=overlaid_server_config, override_configs=override)
+    logger.debug(f'final server config: {overlaid_server_config}')
+    _dict = overlaid_server_config
```

Comment:
What if a config item set previously in SERVER_CONFIG and loaded to memory but get removed later? Can this item be removed by overlay on reloading?

Reply:
Ah, I got it, we just overlay on an empty config